### PR TITLE
Legitimizes BOMBGERs. Add grenades to burgers and assistants will still eat it.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Here's the changes to custom food:
- Can now accept grenades
- If you add something larger than the custom food, the custom food becomes that size
- Biting down on food with grenades in them triggers the grenades

## Why It's Good For The Game

We're trying so hard to remove getting around this whole bombger thing but wait it's actually based as hell. Why don't we just cut the fat of what doesn't work (getting around sanity checks with deep frying) and legitimize what does?

## Changelog
:cl:
add: You can add grenades to food
add: Grenades explode when accidentally bitten into
balance: Adding something large to custom food will make it, too, also large
/:cl:

### CODE REVIEW

**Comment by flowercuco** [comment]
i posted the initial bug report on the server of the downstream i play on in hopes of someone doing it in a round before we merged whatever fix happened so legitimizing it definitely sounds fun

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/components/food/edible.dm [hunk 1]
```diff
@@ -429,27 +429,32 @@ Behavior that's still missing from this component that original food items had t
...
+	//Invoke the eater's stomach's after_eat callback if valid
+	if(iscarbon(eater))
+		var/mob/living/carbon/carbon_eater = eater
+		var/obj/item/organ/internal/stomach/stomach = carbon_eater.getorganslot(ORGAN_SLOT_STOMACH)
+		if(istype(stomach))
+			stomach.after_eat(owner)
+
+
+	for(var/obj/item/grenade/bombger_ingredient in parent)
+		bombger_ingredient.detonate()
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/datums/components/food/edible.dm:455]: shore

**Comment by EuclidianGeometer** [comment]
you can make a few custom foods which are sliceable. what happens then?

**Comment by tralezab** [comment]
> you can make a few custom foods which are sliceable. what happens then?

I'll make them go boom on slice

**Comment by Wallemations** [comment]
Remind me, do custom burgers tell you what's in them if you examine them?

**Comment by Serxule** [comment]
> Remind me, do custom burgers tell you what's in them if you examine them?

only in the name afaik but they can be renamed

**Comment by tralezab** [comment]
> Remind me, do custom burgers tell you what's in them if you examine them?

Name and description, for most things.
image

**Comment by RikuTheKiller** [comment]
Yes, YES, YESSSS!

**Comment by Pickle-Coding** [comment]
I think adding fried TTVs should add a unique overlay, to keep it recognizable and not concealed so easily.

**Comment by RikuTheKiller** [comment]
> I think adding fried TTVs should add a unique overlay, to keep it recognizable and not concealed so easily.

Good idea.

**Comment by Shadowflame909** [comment]
as long as it is made with love i will eat it

clown lube bomb soup soon

**Comment by Overuseddart** [comment]
The Captain after eating my burger with 5 badmin payloads and a c4:

**Comment by datorangebottle** [comment]
> > I think adding fried TTVs should add a unique overlay, to keep it recognizable and not concealed so easily.
> 
> Seconded, a full TTV should at least have some sort of tell compared to a grenade. I think visibly showing a TTV between two buns is the ultimate comedy moment.

Honestly, if someone handed me an extremely obvious TTV burger, I would stand next to them while eating it.

File: code/game/objects/items/grenades/_grenade.dm [hunk 2]
```diff
@@ -163,6 +167,18 @@
...
 		mob.dropItemToGround(src)
 
+/obj/item/grenade/proc/on_used_as_ingredient(datum/source, atom/used_in)
+	SIGNAL_HANDLER
+
+	RegisterSignal(used_in, COMSIG_FOOD_EATEN, PROC_REF(ingredient_detonation))
+	RegisterSignal(used_in, COMSIG_TOOL_ATOM_ACTED_PRIMARY(TOOL_KNIFE), PROC_REF(ingredient_detonation))
+	RegisterSignal(used_in, COMSIG_TOOL_ATOM_ACTED_PRIMARY(TOOL_ROLLINGPIN), PROC_REF(ingredient_detonation))
+
+/obj/item/grenade/proc/ingredient_detonation(datum/source, mob/living/target, mob/living/user, bitecount, bitesize)
```
Comment (Time-Green) [inline code/game/objects/items/grenades/_grenade.dm:177]: Can you doc these?

File: code/game/objects/items/grenades/_grenade.dm [hunk 3]
```diff
@@ -163,6 +167,18 @@
 		var/mob/mob = loc
 		mob.dropItemToGround(src)
 
+/obj/item/grenade/proc/on_used_as_ingredient(datum/source, atom/used_in)
```
Comment (Time-Green) [inline code/game/objects/items/grenades/_grenade.dm:170]: And doc this one?

File: code/datums/components/customizable_reagent_holder.dm [hunk 4]
```diff
@@ -110,6 +110,8 @@
 /datum/component/customizable_reagent_holder/proc/valid_ingredient(obj/ingredient)
 	if (HAS_TRAIT(ingredient, TRAIT_CUSTOMIZABLE_REAGENT_HOLDER))
 		return FALSE
+	if(isgrenade(ingredient))
```
Comment (Time-Green) [inline code/datums/components/customizable_reagent_holder.dm:113]: Could you make this look for a trait instead? TRAIT_BURGERABLE_NON_FOOD or something so this doesn’t devolve into a list of istype checks whenever someone else wants to add make non-food object be addable to food

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by Time-Green** [comment]
Could you also show a picture of how a bombger looks like in-game?

**Comment by Overuseddart** [comment]
Can we stack as many bombs as we can ingredients? I want to stack 99 syndibombs and 1 c4 to see what happens

**Comment by tralezab** [comment]
one frag in the fragburger to the right, 10 frags to the left
image

**Comment by Time-Green** [comment]
> Can you still fire bombgers into someone’s mouth with a pneumatic cannon, causing them to eat it?

new meta

**Pull request merged**